src/org/benf/cfr/reader/bytecode/analysis/parse/expression/HandleExpression.java

```diff
@@ -0,0 +1,76 @@
+package org.benf.cfr.reader.bytecode.analysis.parse.expression;
+
+import org.benf.cfr.reader.bytecode.analysis.parse.*;
+import org.benf.cfr.reader.bytecode.analysis.parse.expression.misc.Precedence;
+import org.benf.cfr.reader.bytecode.analysis.parse.literal.TypedLiteral;
+import org.benf.cfr.reader.bytecode.analysis.parse.rewriters.*;
+import org.benf.cfr.reader.bytecode.analysis.parse.utils.*;
+import org.benf.cfr.reader.bytecode.analysis.types.discovery.InferredJavaType;
+import org.benf.cfr.reader.entities.constantpool.*;
+import org.benf.cfr.reader.util.output.Dumper;
+
+public class HandleExpression extends AbstractExpression {
+  private ConstantPoolEntryMethodHandle handle;
+
+  public HandleExpression(ConstantPoolEntryMethodHandle handle) {
+      super(new InferredJavaType(handle.getDefaultType(), InferredJavaType.Source.FUNCTION, true));
+      this.handle = handle;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null) return false;

```

Comment:
Note that a `null` check here is not needed since the `! instanceof` below covers that.
Though it does not matter that much.